runtime/src/server/middleware/get_page_handler.ts

```diff
@@ -267,7 +267,10 @@ export function get_page_handler(
 			const { html, head, css } = App.render(props);
 
 			const serialized = {
-				preloaded: `[${preloaded.map(data => try_serialize(data)).join(',')}]`,
+				preloaded: `[${preloaded.map(data => try_serialize(data, err => {
```

Comment:
should we use `serialize_error` in the error case?

https://github.com/sveltejs/sapper/blob/427a6bca9a0dbca7fcf3efe0841f63fc50387e84/runtime/src/server/middleware/get_page_handler.ts#L382

Reply:
Like the comment suggests, that (breaking) changes the current behavior from *server failing -> client trying again* to *server failing -> showing persistent error page*. I think an important contributor / maintainer / core developer would have to decide that.

Reply:
I believe we already do that for any other type of failure, so it would be the most consistent behavior. I'm not sure why a serialization failure would be treated any differently

Reply:
Switching to that behavior in c391c9a makes [the `preloading/bails on custom classes returned from preload` test](https://github.com/sveltejs/sapper/blob/master/test/apps/preloading/test.ts#L47-L54) fail.

Reply:
My vote would be to update the test. Perhaps it would be considered a breaking change and we have to wait until 0.28.0 to merge it. Thoughts @antony @Conduitry ?

Reply:
This is certainly a behaviour I would prefer, yes. It is indeed breaking, but in the best possible way.